api/models/metadata.go

```diff
@@ -95,17 +57,17 @@ func (metadata *Metadata) Scan(value interface{}) error {
 
 func (metadata *Metadata) ToLabel() map[string]string {
 	labels := map[string]string{
-		prefix + labelAppName:          metadata.App,
-		prefix + labelComponent:        metadata.Component,
-		prefix + labelEnvironment:      environment,
-		prefix + LabelOrchestratorName: orchestratorValue,
-		prefix + labelStreamName:       metadata.Stream,
-		prefix + labelTeamName:         metadata.Team,
+		labeller.GetLabelName(labeller.LabelAppName):          metadata.App,
+		labeller.GetLabelName(labeller.LabelComponent):        metadata.Component,
+		labeller.GetLabelName(labeller.LabelEnvironment):      labeller.GetEnvironment(),
+		labeller.GetLabelName(labeller.LabelOrchestratorName): orchestratorValue,
+		labeller.GetLabelName(labeller.LabelStreamName):       metadata.Stream,
+		labeller.GetLabelName(labeller.LabelTeamName):         metadata.Team,
 	}
 
 	for _, label := range metadata.Labels {
 		// skip label that is trying to override reserved key
-		if _, usingReservedKeys := reservedKeys[prefix+label.Key]; usingReservedKeys {
+		if labeller.IsReservedKey(labeller.GetPrefix() + label.Key) {
```

Comment:
Hang on if we add the prefix to the label key, e.g. `gojek.com/` + `team`, and look up its result in the `reservedKeys` map, we won't be able to find `gojek.com/team` in the map right? Since the map contains the label key values only i.e. `team`. Was this even working before? 😦

Reply:
You're right, this will only work if the prefix is `""` (but it can't be `""`). Was this working before? I honestly don't think so 😅 but Turing also work the same way, so I just keep the functionality as is

Reply:
Hmm alright I'll create a card for this in the backlog so that we can fix this in the future. Thanks for clarifying!